build-tests/heft-rspack-everything-test/rspack.config.mjs

```diff
@@ -0,0 +1,50 @@
+// @ts-check
+/** @typedef {import('@rushstack/heft-rspack-plugin').IRspackConfiguration} IRspackConfiguration */
+'use strict';
+
+import { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { HtmlRspackPlugin, SwcJsMinimizerRspackPlugin } from '@rspack/core';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+/** @type {IRspackConfiguration} */
+const config = {
+  mode: 'production',
+  module: {
+    rules: [
+      {
+        test: /\.png$/i,
+        type: 'asset/resource'
+      },
+      {
+        test: /\.js$/,
+        enforce: 'pre',
+        use: ['source-map-loader']
+      }
+    ]
+  },
+  target: ['web', 'es2020'],
+  resolve: {
+    extensions: ['.js', '.jsx', '.json']
```

Comment:
`jsx` being in this list is non-standard. The compiler is supposed to handle that.

Reply:
also removing from all build-tests projects